irohad/ametsuchi/impl/mutable_storage_impl.cpp

```diff
@@ -23,48 +22,75 @@ namespace iroha {
         std::shared_ptr<shared_model::interface::CommonObjectsFactory> factory)
         : top_hash_(top_hash),
           sql_(std::move(sql)),
-          wsv_(std::make_shared<PostgresWsvQuery>(*sql_, factory)),
+          peer_query_(std::make_unique<PeerQueryWsv>(
+              std::make_shared<PostgresWsvQuery>(*sql_, factory))),
           block_index_(std::make_unique<PostgresBlockIndex>(*sql_)),
           command_executor_(std::make_shared<PostgresCommandExecutor>(*sql_)),
           committed(false),
           log_(logger::log("MutableStorage")) {
       *sql_ << "BEGIN";
     }
 
-    bool MutableStorageImpl::check(
-        const shared_model::interface::Block &block,
-        MutableStoragePredicate predicate) {
-      PeerQueryWsv peer_query(wsv_);
-      return predicate(block, peer_query, top_hash_);
-    }
-
-    bool MutableStorageImpl::apply(const shared_model::interface::Block &block) {
+    bool MutableStorageImpl::apply(const shared_model::interface::Block &block,
+                                   MutableStoragePredicate function) {
       auto execute_transaction = [this](auto &transaction) {
         command_executor_->setCreatorAccountId(transaction.creatorAccountId());
         command_executor_->doValidation(false);
+
         auto execute_command = [this](auto &command) {
           auto result = boost::apply_visitor(*command_executor_, command.get());
+
           return result.match([](expected::Value<void> &v) { return true; },
                               [&](expected::Error<CommandError> &e) {
                                 log_->error(e.error.toString());
                                 return false;
                               });
         };
+
         return std::all_of(transaction.commands().begin(),
                            transaction.commands().end(),
                            execute_command);
       };
 
-      *sql_ << "SAVEPOINT savepoint_";
-      auto result = std::all_of(block.transactions().begin(),
-                          block.transactions().end(),
-                          execute_transaction);
+      log_->info("Applying block: height {}, hash {}",
+                 block.height(),
+                 block.hash().hex());
+
+      if (not function(block, *peer_query_, top_hash_)) {
+        return false;
+      }
 
+      auto result = std::all_of(block.transactions().begin(),
```

Comment:
Don't like name `result` for a `bool` variable. Consider changing to, for example, `block_applied_successfully` or something like this